### PR TITLE
Dependabot/maven/aws react jwt assembly GitHub

### PR DESCRIPTION
**Pull request by sb-keane**
**JIRA Tickets:**

[AB2D-4823](https://jira.cms.gov/browse/AB2D-4823) - Description
Combine dependabot PRs and test them. 

### What Does This PR Do?
Combines these PRs
https://github.com/CMSgov/ab2d/pull/1062
https://github.com/CMSgov/ab2d/pull/1060
https://github.com/CMSgov/ab2d/pull/1059
https://github.com/CMSgov/ab2d/pull/1051

Revert docker compose file version to 2.4 to fix errors caused by the use of deprecated features. V3 no longer supports  depend_on conditions. 

Remove reactor-spring dependency which is not needed and is quite old. 

**Revert api and worker java docker image from openjdk:19 back to openjdk:17. 
openjdk:17 has a few more security issues but New Relic doesn't work with 19 yet.** 

Specify Postgress driver version in the app pom and update the module poms to use that version. 

### What Should Reviewers Watch For?

Unintended issues for individual local environments. 

### Database Changes

- [ ] All QuickSight impacts have been identified

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [X] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [X] Code checked for PHI/PII exposure

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: '3'
+version: '2.4'
```
Comment (sb-keane) [inline docker-compose.yml:1]: Docker deprecated depend_on ->service -> condition in v3. There is a replacement but that only works with swarms. 
https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on

File: hpms/pom.xml [hunk 2]
```diff
@@ -50,11 +50,7 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-webflux</artifactId>
         </dependency>
-        <dependency>
```
Comment (sb-keane) [inline hpms/pom.xml:53]: This dependency isn't used and pulls in dependencies that are insecure. 
https://mvnrepository.com/artifact/org.projectreactor/reactor-spring/1.0.1.RELEASE

File: api/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM openjdk:19
+FROM openjdk:17
```
Comment (sb-keane) [inline api/Dockerfile:1]: New Relic doesn't work with 19 yet.

File: eventlogger/pom.xml [hunk 4]
```diff
@@ -73,6 +73,7 @@
         <dependency>
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
+            <version>42.4.0</version>
```
Comment (sharshar-sb) [inline eventlogger/pom.xml:76]: It looks like the version is not specified any any of the postgresql dependencies. Create a variable in the top level pom for the version and propagate it down to all the poms.

**Review by sharshar-sb — CHANGES_REQUESTED** [review]

File: .github/workflows/combine-prs.yml [hunk 5]
```diff
@@ -122,7 +122,7 @@ jobs:
           git pull origin $sourcebranches --no-edit
           git push origin $COMBINE_BRANCH_NAME
       # Creates a PR with the new combined branch
-      - uses: actions/github-script@v3
+      - uses: actions/github-script@v5
```
[the remark this answers is not in the document]
Comment (sb-keane) [reply .github/workflows/combine-prs.yml:125]: That was a typo. Thanks for catching it!

**Review by sharshar-sb — APPROVED** [review]

**Review by sb-dkniesteadt — APPROVED** [review]

**Pull request merged**